src/connector/src/source/iceberg/mod.rs

```diff
@@ -302,10 +308,12 @@ impl IcebergSplitEnumerator {
                 .files
                 .push(data_files[split_num * split_size + i].clone());
         }
-        Ok(splits
+        let splits = splits
             .into_iter()
             .filter(|split| !split.files.is_empty())
-            .collect_vec())
+            .collect_vec();
+
+        Ok(splits)
```

Comment:
After the predicate pushdown, it is possible that the splits become empty. If we don't handle splits properly, it will cause a panic. Let return an split with empty files in this case
```
return Ok(vec![IcebergSplit {
        split_id: 0,
        snapshot_id: 0,
        table_meta: TableMetadataJsonStr::serialize(table.metadata()),
        files: vec![],
        equality_delete_files: vec![],
        position_delete_files: vec![],
    }]);
```

Reply:
https://github.com/risingwavelabs/risingwave/pull/19228/commits/bb07b036a61e7621f9a3f8c76dad743c49feaaf7

Fixed.

I previously handled this by creating an empty executor instead.